### PR TITLE
API-2320: R4 Appointment model

### PR DESCRIPTION
**Pull request by jjabando1**
Appointment, AppointmentTest, and SampleAppointments classes initialized

### CODE REVIEW

File: r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+  // Ancestor -- DomainResource
+  @Valid Narrative text;
+  @Valid List<SimpleResource> contained;
+  @Valid List<Extension> extension;
+  @Valid List<Extension> modifierExtension;
+
+  // Appointment Resource
+  @Valid List<Identifier> identifier;
+  @Valid @NotNull AppStatus status;
```
Comment (ilaflamme-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java:69]: AppointmentStatus

File: r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+  @SuppressWarnings("unused")
+  public enum Required {
+    required,
+    optional,
+    @JsonProperty("information-only")
+    information_only
+  }
+
+  @SuppressWarnings("unused")
+  public enum PartStatus {
```
Comment (ilaflamme-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java:139]: ParticipationStatus

File: r4/src/test/java/gov/va/api/health/r4/api/samples/SampleAppointments.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        .patientInstruction("Sample Appt Patient Instruction")
+        .basedOn(singletonList(reference()))
+        .participant(singletonList(participant()))
+        .requestedPeriod(singletonList(period()))
+        .build();
+  }
+
+  public Participant participant() {
+    return Participant.builder()
+        .actor(reference())
```
Comment (ilaflamme-va) [inline r4/src/test/java/gov/va/api/health/r4/api/samples/SampleAppointments.java:57]: missing type

**Comment by ilaflamme-va** [comment]
Small, nitpicks, but otherwise goodjob!

File: r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+  @Data
+  @NoArgsConstructor
+  @EqualsAndHashCode(callSuper = true)
+  @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
+  @JsonDeserialize(builder = Appointment.Bundle.BundleBuilder.class)
+  @Schema(
+      name = "AppointmentBundle",
+      example =
+          "${r4.apptBundle:gov.va.api.health.r4.api.swaggerexamples.SwaggerAppointment#apptBundle}")
```
Comment (eclendenning-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java:155]: appointmentBundle

File: r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+  @Valid CodeableConcept cancelationReason;
+  @Valid List<CodeableConcept> serviceCategory;
+  @Valid List<CodeableConcept> serviceType;
+  @Valid List<CodeableConcept> specialty;
+  @Valid CodeableConcept appointmentType;
+  @Valid List<CodeableConcept> reasonCode;
+  @Valid List<Reference> reasonReference;
+
+  @Min(0)
+  @Valid
```
Comment (jhulbert-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java:79]: Doesnt need the valid annotation

File: r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+  @Valid List<Reference> supportingInformation;
+
+  @Pattern(regexp = Fhir.INSTANT)
+  String start;
+
+  @Pattern(regexp = Fhir.INSTANT)
+  String end;
+
+  @Min(1)
+  @Valid
```
Comment (jhulbert-va) [inline r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java:94]: Doesnt need the valid annotation

**Review by eclendenning-va — CHANGES_REQUESTED** [review]

File: r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java (hunk 4 above)
Comment (jjabando1) [reply r4/src/main/java/gov/va/api/health/r4/api/resources/Appointment.java:155]: I set the example string to "${r4.appointmentBundle:gov.va.api.health.r4.api.swaggerexamples.SwaggerAppointment#appointmentBundle}", but I get a build error: src\main\java\gov\va\api\health\r4\api\resources\Appointment.java:[153] (sizes) LineLength: Line is longer than 100 characters (found 114).

**Review by eclendenning-va — APPROVED** [review]

**Pull request merged**